packages/legend-extension-external-language-morphir/src/ELMorphir_GenerationPlugin.tsx

```diff
@@ -0,0 +1,173 @@
+}
+
+export class ELMorphir_GenerationPlugin
+  extends StudioPlugin
+  implements DSL_StudioPlugin_Extension
+{
+  networkClient: NetworkClient;
+  morphirVisualizerUrl = ``;
+  linterServerUrl = ``;
+  linterAppUrl = ``;
+
+  constructor() {
+    super(packageJson.extensions.studioPlugin, packageJson.version);
+    this.networkClient = new NetworkClient();
+  }
+
+  install(pluginManager: StudioPluginManager): void {
+    pluginManager.registerStudioPlugin(this);
+  }
+
+  override configure(_configData: object): ELMorphir_GenerationPlugin {
+    const configData = _configData as ELMorphir_GenerationPluginConfigData;
+    assertNonEmptyString(
+      configData.morphirVisualizerUrl,
+      `Can't configure morphir visualizer url for generation plugin: 'url' field is missing or empty`,
+    );
+    assertNonEmptyString(
+      configData.linterServerUrl,
+      `Can't configure linter server url for generation plugin: 'url' field is missing or empty`,
+    );
+    assertNonEmptyString(
+      configData.linterAppUrl,
+      `Can't configure linter app url for generation plugin: 'url' field is missing or empty`,
+    );
+    this.morphirVisualizerUrl = configData.morphirVisualizerUrl;
+    this.linterServerUrl = configData.linterServerUrl;
+    this.linterAppUrl = configData.linterAppUrl;
+
+    return this;
+  }
+
+  visualizeMorphir =
+    (fileNode: GenerationFile): (() => void) =>
+    async (): Promise<void> => {
+      assertNonEmptyString(this.morphirVisualizerUrl);
+      window.open(this.morphirVisualizerUrl);
+      await this.networkClient.post(
+        this.morphirVisualizerUrl,
+        fileNode.content,
+      );
+    };
+
+  visualizeBosque =
+    (
+      fileGenerationState: FileGenerationState,
+      fileNode: GenerationFile,
+    ): (() => void) =>
+    async (): Promise<void> => {
+      assertNonEmptyString(this.linterServerUrl);
+      assertNonEmptyString(this.linterAppUrl);
+      const code =
+        fileGenerationState.editorStore.graphManagerState.graphManager.graphToPureCode(
+          fileGenerationState.editorStore.graphManagerState.graph,
+        );
+      await this.networkClient.post(this.linterServerUrl, {
+        ir: fileNode.content,
+        src: await code,
+      });
+      window.open(this.linterAppUrl);
+    };
+
+  getExtraFileGenerationResultViewerActions(): FileGenerationResultViewerAction[] {
+    return [
+      (
+        fileGenerationState: FileGenerationState,
+        fileNode: GenerationFile,
+      ): React.ReactNode | undefined => {
+        if (this.isMorphirGenerationType(fileGenerationState)) {
+          return (
+            <div className="panel__header__title__content generation-result-viewer__file__header-button">
+              <button
+                className="panel__content__form__section__list__new-item__add-btn btn btn--dark"
+                onClick={this.visualizeMorphir(fileNode)}
+                tabIndex={-1}
+              >
+                Visualize Generated IR
```

Comment:
Are you sure you don't want to use `icons` here instead? If you use icon though, you should have these as the titles (in `Sentence case`)

Reply:
I think I want to maintain the literal button now for clarity.